File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def dfs(i, visited, c):
    visited[i] = 1

    for idx, x in enumerate(c[i]):
        if not visited[idx] and x:
            dfs(idx, visited, c)

    return

def solution(n, computers):
    answer = 0
    visited = [0] * n

    for i in range(n):
        if not visited[i]:
            dfs(i, visited, computers)
            answer += 1

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[201];

void dfs(int i, int n, vector<vector<int>> computers){
    visited[i] = 1;
    
    for(int j=0;j<n;j++){
        if(!visited[j] && computers[i][j] == 1){
            dfs(j,n,computers);
        }
    }
    return;
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    for(int i=0;i<n;i++){
        if(!visited[i]){
            dfs(i,n,computers);
            answer ++;
        }
    }
    return answer;
}